File: rock_paper_scissors/rps.py
#!/usr/bin/python

import sys

def rock_paper_scissors(n):
    if n == 0:
        return [[]]
    elif n == 1:
        return [['rock'], ['paper'], ['scissors']]
    else:
        plays = rock_paper_scissors(1)
        for i in range(n-1):
            new_plays = []
            for move_list in plays:
                for move in rock_paper_scissors(1):
                    new_plays += [move_list + move]
            plays = new_plays
        return plays


if __name__ == "__main__":
    if len(sys.argv) > 1:
        num_plays = int(sys.argv[1])
        print(rock_paper_scissors(num_plays))
    else:
        print('Usage: rps.py [num_plays]')

File: rock_paper_scissors/rps.js
function rock_paper_scissors(n) {
  rps_list = [['rock'], ['paper'], ['scissors']];

  if (n == 0) {
    return [[]];
  }

  if (n == 1) {
    return list;
  }

  combo = [];

  for (let i = 0; i < 3 ** n; i++) {
    if (i < Math.ceil(0.3333 * 3 ** n)) {
      combo.push(rps_list[0]);
    } else if (i < Math.ceil(0.6666 * 3 ** n)) {
      combo.push(rps_list[1]);
    } else {
      combo.push(rps_list[2]);
    }
  }

  while (combo[0].length !== n)
    for (let i = 0; i < 3 ** n; i++) {
      combo[i] = [...combo[i], ...rps_list[i % 3]];
    }

  //   combo[0] += rps_list[0];
  return combo;
}

console.log(rock_paper_scissors(3));
